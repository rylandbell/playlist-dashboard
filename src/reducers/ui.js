export const hoveredTrackRow = (state = null, action) => {
  switch (action.type){
    case 'HOVER_ON_TRACK':
      return action.data;
    case 'CLEAR_HOVERED_TRACK':
      return null
    default:
      return state;
  }
}

export const activeSidebarTab = (state = "playlists", action) => {
  switch (action.type) {
    case 'SET_ACTIVE_TAB':
      return action.data;
    case 'SELECT_PLAYLIST':
      return action.forceTabSwitch ? "filters" : state;
    default:
      return state;
  }
}

// on first playlist select, automatically push user to Filters tab. after that, let user control tabs
export const autoSidebarTabSwitch = (state = true, action) => {
  switch (action.type) {
    case 'SELECT_PLAYLIST':
      return false;
    default:
      return state;
  }
}

export const newPlaylistName = (state = "", action) => {
  switch (action.type) {
    case 'SELECT_PLAYLIST':
      return "Modified: " + action.data.name;
    case 'CHANGE_NAME_TEXT':
      return action.data;
    default:
      return state;
  }
}

export const animateNextChartDraw = (state = true, action) => {
  switch (action.type) {
    case 'STOP_ANIMATING_CHART':
      return false;
    case 'ADD_AUDIO_FEATURES':
      return true;
    default:
      return state;
  }
}