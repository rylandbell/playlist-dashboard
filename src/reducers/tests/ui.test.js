import * as reducers from '../ui';

describe('UI reducers', () => {
  describe('hoveredTrackId', () => {
    const initState = reducers.hoveredTrackId(undefined, { type: 'INIT' });

    it('should return the ID of the hovered track', () => {
      const action = { type: 'HOVER_ON_TRACK', data: { id: 'test' } };
      expect(reducers.hoveredTrackId(initState, action)).toMatchSnapshot();
    });

    it('should return null with CLEAR_HOVERED_TRACK action type', () => {
      const action = { type: 'CLEAR_HOVERED_TRACK' };
      expect(reducers.hoveredTrackId(initState, action)).toMatchSnapshot();
    });
  });

  describe('activeSidebarTab', () => {
    const initState = reducers.activeSidebarTab(undefined, { type: 'INIT' });

    it("should return the action's data for SET_ACTIVE_TAB actions", () => {
      const action = { type: 'SET_ACTIVE_TAB', data: 'test' };
      expect(reducers.activeSidebarTab(initState, action)).toMatchSnapshot();
    });

    it('should return "filters" for SELECT_PLAYLIST actions with forceTabSwitch ', () => {
      const action = { type: 'SELECT_PLAYLIST', forceTabSwitch: true };
      expect(reducers.activeSidebarTab(initState, action)).toMatchSnapshot();
    });

    it('should return the state for SELECT_PLAYLIST actions without forceTabSwitch ', () => {
      const action = { type: 'SELECT_PLAYLIST' };
      expect(reducers.activeSidebarTab(initState, action)).toMatchSnapshot();
    });
  });

  describe('autoSidebarTabSwitch', () => {
    const initState = reducers.autoSidebarTabSwitch(undefined, {
      type: 'INIT'
    });

    it('should always return false for SELECT_PLAYLIST actions', () => {
      const action = { type: 'SELECT_PLAYLIST' };
      expect(reducers.autoSidebarTabSwitch(initState, action)).toEqual(false);
    });
  });

  describe('newPlaylistName', () => {
    const initState = reducers.newPlaylistName(undefined, { type: 'INIT' });

    it('should return action data for CHANGE_NAME_TEXT actions', () => {
      const action = { type: 'CHANGE_NAME_TEXT', data: { name: 'test' } };
      expect(reducers.newPlaylistName(initState, action)).toMatchSnapshot();
    });

    it('should prepend "Modified: " to action data for SELECT_PLAYLIST actions', () => {
      const action = { type: 'SELECT_PLAYLIST', data: { name: 'test' } };
      expect(reducers.newPlaylistName(initState, action)).toMatchSnapshot();
    });
  });

  describe('animateNextChartDraw', () => {
    const initState = reducers.animateNextChartDraw(undefined, {
      type: 'INIT'
    });

    it('should return false for STOP_ANIMATING_CHART actions', () => {
      const action = { type: 'STOP_ANIMATING_CHART' };
      expect(reducers.animateNextChartDraw(initState, action)).toEqual(false);
    });

    it('should return true for ADD_FEATURES_DATA actions', () => {
      const action = { type: 'ADD_FEATURES_DATA' };
      expect(reducers.animateNextChartDraw(initState, action)).toEqual(true);
    });
  });
});
