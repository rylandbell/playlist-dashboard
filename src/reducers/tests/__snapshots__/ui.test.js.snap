// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI reducers activeSidebarTab should return "filters" for SELECT_PLAYLIST actions with forceTabSwitch  1`] = `"filters"`;

exports[`UI reducers activeSidebarTab should return the action's data for SET_ACTIVE_TAB actions 1`] = `"test"`;

exports[`UI reducers activeSidebarTab should return the state for SELECT_PLAYLIST actions without forceTabSwitch  1`] = `"playlists"`;

exports[`UI reducers hoveredTrackId should return null with CLEAR_HOVERED_TRACK action type 1`] = `null`;

exports[`UI reducers hoveredTrackId should return the ID of the hovered track 1`] = `"test"`;

exports[`UI reducers newPlaylistName should prepend "Modified: " to action data for SELECT_PLAYLIST actions 1`] = `"Modified: test"`;

exports[`UI reducers newPlaylistName should return action data for CHANGE_NAME_TEXT actions 1`] = `
Object {
  "name": "test",
}
`;
